include/GoogleSync/GoogleSyncBase.php

```diff
@@ -0,0 +1,1049 @@
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Make sure we have a calendar id
+     *
+     * @return boolean
+     */
+    protected function isCalendarExists()
+    {
+        // Make sure we have a calendar id
+        if (!isset($this->calendarId)) {
+            // TODO: I think a warning enough into log instead fatal (but it should be handled by caller function)
+            $this->logger->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'The calendar ID is not set. See setUsersGoogleCalendar Method');
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Get a google event by the event id
+     *
+     * @param string $event_id Google Event ID
+     *
+     * @return \Google_Service_Calendar_Event|null Google_Service_Calendar_Event if found, null if not found
+     * @throws GoogleSyncException if $event_id is empty
+     * @throws GoogleSyncException if Google Service not set up
+     */
+    protected function getGoogleEventById($event_id)
+    {
+
+        if (empty($event_id)) {
+            // If we didn't get passed an event id, throw an exception
+            throw new GoogleSyncException('event ID is empty', GoogleSyncException::EVENT_ID_IS_EMPTY);
+        }
+
+        // Make sure the calendar service is set up
+        if (!$this->isServiceExists()) {
+            throw new GoogleSyncException('Cannot Continue Without Google Service', GoogleSyncException::GCALENDAR_FAILURE);
+        }
+
+        $gEvent = $this->gService->events->get($this->calendarId, $event_id);
+
+        if (!empty($gEvent->getId())) {
+            return $gEvent;
+        }
+        return null;
+    }
+
+    /**
+     * Get a SuiteCRM meeting by Google Event ID
+     *
+     * @param string $event_id The Google Event ID
+     *
+     * @return \Meeting|null SuiteCRM Meeting Bean if found, null if not found
+     * @throws GoogleSyncException if more than one meeting matches $event_id
+     * @throws GoogleSyncException If unable to retrieve meeting bean
+     */
+    protected function getMeetingByEventId($event_id)
+    {
+
+        // We do it this way so we also get deleted meetings
+        $eventIdQuoted = $this->db->quoted($event_id);
+        $query = "SELECT id FROM meetings WHERE gsync_id = {$eventIdQuoted}";
+        $result = $this->db->query($query);
+        
+        if (!$result) {
+            throw new GoogleSyncException('Meeting not found with specified gsync_id: ' . $eventIdQuoted, GoogleSyncException::MEETING_NOT_FOUND);
+        }
+
+        // This checks to make sure we only get one result. If we get more than one, something is inconsistant in the DB
+        if ($result->num_rows > 1) {
+            // TODO: It shouldn't be in the log as it is an exception, it will be logged if unhandled by suitecrm core.
```

Comment:
@gymad
I'll clean this up... but I want to make it clear that this is a 'scream from the hilltops' error. More then one row returned means something has gone horribly wrong with the data in the DB. Corruption.

Reply:
Fixed